sqlglot/dialects/clickhouse.py

```diff
@@ -870,6 +899,14 @@ def cast_sql(self, expression: exp.Cast, safe_prefix: t.Optional[str] = None) ->
 
             return super().cast_sql(expression, safe_prefix=safe_prefix)
 
+        def trycast_sql(self, expression: exp.TryCast) -> str:
+            target_type = self.sql(expression.to)
+            if target_type.upper() in self.TRY_CAST_TYPES:
```

Comment:
will this never be non upper?

Reply:
Yea, because CH has some case-sensitive types like `UInt8`, but some case-insensitive ones too like `DATE`.